R/graph.R

```diff
@@ -45,73 +87,389 @@ writeReactLog <- function(file=stdout(), sessionToken = NULL) {
 #' time that each reactive.
 #' @export
 showReactLog <- function(time = TRUE) {
-  utils::browseURL(renderReactLog(time = as.logical(time)))
+  check_reactlog()
+  reactlog::show_reactlog(rLog$asList(), time = time)
 }
-
+# function should not be called. only keeping for legacy purposes
 renderReactLog <- function(sessionToken = NULL, time = TRUE) {
-  templateFile <- system.file('www/reactive-graph.html', package='shiny')
-  html <- paste(readLines(templateFile, warn=FALSE), collapse='\r\n')
-  tc <- textConnection(NULL, 'w')
-  on.exit(close(tc))
-  writeReactLog(tc, sessionToken)
-  cat('\n', file=tc)
-  flush(tc)
-  html <- sub('__DATA__', paste(textConnectionValue(tc), collapse='\r\n'), html, fixed=TRUE)
-  html <- sub('__TIME__', paste0('"', time, '"'), html, fixed=TRUE)
-  file <- tempfile(fileext = '.html')
-  writeLines(html, file)
-  return(file)
+  check_reactlog()
+  reactlog::render_reactlog(
+    rLog$asList(),
+    session_token = sessionToken,
+    time = time
+  )
+}
+# function should not be called. only keeping for legacy purposes
+#' @importFrom utils getFromNamespace
+writeReactLog <- function(file=stdout(), sessionToken = NULL) {
+  check_reactlog()
+  reactlog::write_reactlog(
+    rLog$asList(),
+    file = file,
+    session_token = sessionToken
+  )
+}
+check_reactlog <- function() {
+  check_suggested("reactlog", "0.0.0.9000", "rstudio/reactlog")
 }
 
-.graphAppend <- function(logEntry, domain = getDefaultReactiveDomain()) {
-  if (isTRUE(getOption('shiny.reactlog'))) {
-    sessionToken <- if (is.null(domain)) NULL else domain$token
-    .graphStack$push(c(logEntry, list(
-      session = sessionToken,
-      time = as.numeric(Sys.time())
-    )))
-  }
 
-  if (!is.null(domain)) {
-    domain$reactlog(logEntry)
-  }
-}
+RLog <- R6Class(
+  "RLog",
+  portable = FALSE,
```

Comment:
@wch is portable or non-portable preferred?

Reply:
If items are not inherited, I think portable doesn't matter.

If I'm not using `<<-` within `RLog` (and always using `self` or `private`, then I won't run into portable / non-portable issues.